openpype/hosts/max/plugins/publish/extract_redshift_proxy.py

```diff
@@ -0,0 +1,77 @@
+import os
+import pyblish.api
+from openpype.pipeline import (
+    publish,
+    OptionalPyblishPluginMixin
+)
+from pymxs import runtime as rt
+from openpype.hosts.max.api import (
+    maintained_selection
+)
+
+
+class ExtractRedshiftProxy(publish.Extractor,
+                           OptionalPyblishPluginMixin):
+    """
+    Extract Camera with AlembicExport
+    """
+
+    order = pyblish.api.ExtractorOrder - 0.1
+    label = "Extract RedShift Proxy"
+    hosts = ["max"]
+    families = ["redshiftproxy"]
+
+    def process(self, instance):
+        container = instance.data["instance_node"]
+        start = int(instance.context.data.get("frameStart"))
+        end = int(instance.context.data.get("frameEnd"))
+
+        self.log.info("Extracting Redshift Proxy...")
+        stagingdir = self.staging_dir(instance)
+        rs_filename = "{name}.rs".format(**instance.data)
+
+        rs_filepath = os.path.join(stagingdir, rs_filename)
+
+        # MaxScript command for export
+        export_cmd = (
+            f"""
+fn ProxyExport fp selected:true compress:false connectivity:false startFrame: endFrame: camera:undefined warnExisting:true transformPivotToOrigin:false = (
+	if startFrame == unsupplied then (
+		startFrame = (currentTime.frame as integer)
+	)
+
+	if endFrame == unsupplied then (
+		endFrame = (currentTime.frame as integer)
+	)
+
```

Comment:
This would never be the case, yes? Since we're passing `start` and `end` which are forced to `int` type?

Reply:
i am still working on that so i just put the codes here first. I will do some adjustments.